src/lib.rs

```diff
@@ -864,7 +865,7 @@ impl SeedableRng for StdRng {
 }
 
 /// An RNG recommended when small state, cheap initialization and good
-/// performance are required. The PRNG algorithm in `SmallRng` is choosen to be
+/// performance are required. The PRNG algorithm in `SmallRng` is chosen to be
 /// efficient on the current platform, **without consideration for cryptography
```

Comment:
True, sometimes it will be about half the price, sometimes it just makes no difference.

And you are getting bitten again by the rust bug with multiple codegen units and benchmarks harness. Can you retry with `export RUSTFLAGS="-C codegen-units=1"`?

Reply:
Interesting; didn't affect most tests (including xorshift bytes with around 900MB/s) but:
```
test gen_u32_xorshift      ... bench:       1,045 ns/iter (+/- 59) = 3827 MB/s
```

Reply:
Actually, that was with another change calling `black_box` less frequently. Without that I get approx 3000 MB/s. The `u64` results only change by about 50MB/s however.